### PR TITLE
Setup post sign-up route

### PR DESCRIPTION
**Pull request by artemruts**
Fixes: https://github.com/sourcegraph/sourcegraph/issues/20342

### Description

The idea behind these changes:
* add new `post-sign-up` route only user's with a `AllowUserViewPostSignup` tag can review
* for the wider roll-out of the post sign-up flow we can use `experimentalFeatures.enablePostSignupFlow` and enable it through "Site configuration" Admin UI
* Stub out a `PostSignUpPage` page

I want this to go out so the team can add new components to that route and we can preview and test them live.

@ryanslade Ryan, is it ok to use `experimentalFeatures` like that?

### CODE REVIEW

File: client/web/src/auth/SignInSignUpCommon.tsx [hunk 1]
```diff
@@ -67,10 +67,15 @@ export const UsernameInput: React.FunctionComponent<InputProps> = props => {
  *
  * 🚨 SECURITY: We must disallow open redirects (to arbitrary hosts).
  */
-export function getReturnTo(location: H.Location): string {
+export function getReturnTo(location: H.Location, isPostSignupEnabled?: boolean): string {
+    if (isPostSignupEnabled) {
+        return new URL('/post-sign-up', window.location.href).pathname
+    }
```
Comment (tjkandala) [inline client/web/src/auth/SignInSignUpCommon.tsx:73]: Will this always return `'/post-sign-up'`? If so, does this need to be inside of `getReturnTo`?

**Review by tjkandala — APPROVED** [review]
web stuff looks good!

**Comment by ryanslade** [comment]
@artemruts Yep, I think that's a good use for `expterimentalFeatures`.

I assume at some point this will become the default in which case we can deprecate the config and stop checking it in code right?

**Comment by artemruts** [comment]
> @artemruts Yep, I think that's a good use for `expterimentalFeatures`.
> 
> I assume at some point this will become the default in which case we can deprecate the config and stop checking it in code right?

Yes, the idea to have it for security now, so we don't enable for everyone by accident. Later, we'll remove it.

**Pull request merged**